core/kogitobuild/build_config.go

```diff
@@ -17,245 +17,41 @@ package kogitobuild
 import (
 	"github.com/kiegroup/kogito-operator/api"
 	"github.com/kiegroup/kogito-operator/core/framework"
-	"github.com/kiegroup/kogito-operator/core/operator"
+	"github.com/kiegroup/kogito-operator/core/framework/util"
 	buildv1 "github.com/openshift/api/build/v1"
-	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"strconv"
-	"strings"
 )
 
 const (
 	// LabelKeyBuildType identifies the instance build type
 	LabelKeyBuildType = "buildType"
-
-	kindImageStreamTag = "ImageStreamTag"
-	tagLatest          = "latest"
-
-	destinationDir   = "."
-	runnerSourcePath = operator.KogitoHomeDir + "/bin"
-
-	nativeBuildEnvVarKey        = "NATIVE"
-	builderLimitCPUEnvVarKey    = "LIMIT_CPU"
-	builderLimitMemoryEnvVarKey = "LIMIT_MEMORY"
-	mavenMirrorURLEnvVar        = "MAVEN_MIRROR_URL"
-	mavenGroupIDEnvVar          = "PROJECT_GROUP_ID"
-	mavenArtifactIDEnvVar       = "PROJECT_ARTIFACT_ID"
-	mavenArtifactVersionEnvVar  = "PROJECT_VERSION"
-	mavenDownloadOutputEnvVar   = "MAVEN_DOWNLOAD_OUTPUT"
-	binaryBuildEnvVar           = "BINARY_BUILD"
 )
 
-// DecoratorHandler ...
-type DecoratorHandler interface {
-	decoratorForRemoteSourceBuilder() decorator
-	decoratorForLocalSourceBuilder() decorator
-	decoratorForSourceBuilder() decorator
-	decoratorForBinaryRuntimeBuilder() decorator
-	decoratorForSourceRuntimeBuilder() decorator
-	decoratorForRuntimeBuilder() decorator
+// BuildConfigHandler ...
+type BuildConfigHandler interface {
+	NewBuildConfig(build api.KogitoBuildInterface, customLabel map[string]string, decorators ...decorator) buildv1.BuildConfig
```

Comment:
Would it be possible and/or meaningful to implement the custom label addition as a decorator?

Reply:
Thanks @sutaakar for point it out.
Sure, we could do so. I will implement a new CustomLabelDecorator for it.

Reply:
done